### PR TITLE
hide unneeded logging messages

### LINKED ISSUE #545: Warnings in recursive-aggregation
When using the recursive aggregation with one of the two lines
```python
data_IIASA.aggregate('Capacity', recursive='skip-validate', append=True)
data_IIASA.aggregate('Capacity', recursive=True, append=True)
```
I most of the time get a (or lots of) warning(s)
```
WARNING: Filtered IamDataFrame is empty!
```
either from pyam.core or from pyam.utils.

Functionally it's not a problem, but it's quite annoying as it fills up the output and makes it hard to see any other - potentially important - print statements. So it would be nice to either hide the warning completely when using recursive aggregation or to generally add a keyword in the aggregate-function to hide these warnings.

Or, one could look deeper into it and see if the recursive-aggregation is doing unnecessary aggregations and eliminate them.

### PR DESCRIPTION
**Pull request by pjuergens**
# Please confirm that this PR has done the following:

~~- [ ] Tests Added~~
~~- [ ] Documentation Added~~
~~- [ ] Name of contributors Added to AUTHORS.rst~~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Fixed unnecessary internal logging messages when aggregating recursively, see Issue #545 

closes #545

### CODE REVIEW

File: pyam/logging.py [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 
 @contextmanager
-def adjust_log_level(logger, level="ERROR"):
+def adjust_log_level(logger="pyam.core", level="ERROR"):
```
Comment (danielhuppmann) [inline pyam/logging.py:9]: I'd rather use "pyam" (ie all pyam modules) as the default rather than just core.

**Comment by danielhuppmann** [comment]
Nice work, thanks @pjuergens!

Two requests:
- Please change the description of this PR to use the GitHub-keyword closes, ie add the phrase "closes #545". Then, GitHub will automatically close the related issue when this PR is merged.
- (optional) Using an underscore in the module name was a bad idea. Could you refactor the module name from `_aggregate` to `aggregation` in this PR?

**Comment by pjuergens** [comment]
Well, most of the work was yours :) I did the refactoring and hope that all the tests will run through now :)

**Comment by pjuergens** [comment]
@danielhuppmann do you know why some tests are failing now? They seem to be in parts of the code where I didn't change anything

**Comment by danielhuppmann** [comment]
> Well, most of the work was yours :) I did the refactoring and hope that all the tests will run through now :)

Well, I think this is a good mode of operation: you (or other intermediate-expert users) identify an issue, receive some guidance from admins like me, then make a PR fixing this...

About the issue, the IIASA database crashed before lunch, so the unit tests checking the **iiasa.Connection** module failed. I restarted the tests, everything passes now. Ready for review?

PS: there is a skip-if-IIASA-db-API decorator if it cannot be reached entirely, but I guess this doesn't when the database times out (rather than being offline)... 
https://github.com/IAMconsortium/pyam/blob/0989fddc03fc88410a31b0d4ea7dff2ef50c3bda/tests/conftest.py#L17

**Review by danielhuppmann — APPROVED** [review]
Thanks @pjuergens for doing some clean-up of the codebase beyond your immediate issue!

**Pull request merged**